tests/generation/test_logits_process.py

```diff
@@ -512,6 +513,27 @@ def test_no_bad_words_dist_processor(self):
         filtered_scores = no_bad_words_dist_proc(input_ids, scores.clone())
         self.assertTrue(torch.allclose(scores, filtered_scores, atol=1e-3))
 
+    def test_bias_dist_processor(self):
+        vocab_size = 5
+        batch_size = 2
+
+        input_ids = torch.tensor([[0, 1, 3, 1], [0, 1, 0, 1]], device=torch_device, dtype=torch.long)
+        positive_bias = {(1,): 100.0, (4,): 100.0}
+        negative_bias = {(1, 0): -100.0, (0, 1, 2): -100.0, (1, 3, 1, 3): -100.0}
+        sequence_bias = {**positive_bias, **negative_bias}
+
+        # scores = 0 to facilitate checks
+        scores = torch.zeros((batch_size, vocab_size), dtype=torch.float, device=torch_device)
+
+        bias_dist_proc = SequenceBiasLogitsProcessor(sequence_bias=sequence_bias)
+        filtered_scores = bias_dist_proc(input_ids, scores.clone())
+
+        # batch 1: positive bias: tokens (1, 4); negative bias: tokens (0, 3); neutral: tokens (2)
+        # batch 2: positive bias: tokens (1, 4); negative bias: tokens (0, 2); neutral: tokens (3)
```

Comment:
I don't understand this comment 😅 

For batch 1, I thought this would be the first row of `input_ids`, but it doesn't contain tokens `4` or `2`. 

Could you explain where e.g. `negative bias: tokens (0, 3);` comes from?

Reply:
For the 1st batch, whose input_ids is `[0, 1, 3, 1]`:
- Tokens `1` and `4` have a positive bias because the bias condition is for a single token, so it's always active;
- Token `0` has a negative bias because the prefix (`input_ids`) ends in `1`, so the first negative bias condition (`(1, 0): -100.0`) is triggered
- Token `3` has a negative bias because the prefix (`input_ids`) ends in `1, 3, 1`, so the third negative bias condition (`(1, 3, 1, 3): -100.0`) is triggered
- All other tokens are neutral

I hope it is clearer now 🙌 (I remember that I had the same questions the 1st time I looked into the tests for the `NoBadWordsLogitsProcessor` haha)